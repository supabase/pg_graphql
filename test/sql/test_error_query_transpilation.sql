begin;

    comment on schema public is '@graphql({"inflect_names": true})';

    create table public.account(
      id serial primary key,
      first_name varchar(255) not null
    );

    insert into public.account(first_name) values ('foo');

    -- Extend with function
    create function public._raise_err(rec public.account)
      returns text
      immutable
      strict
      language sql
    as $$
      select 1/0 -- divide by 0 error
    $$;

    select
      jsonb_pretty(
        graphql.resolve($$
          {
            accountCollection {
              edges {
                node {
                  id
                  firstName
                  raiseErr
                }
              }
            }
          }
        $$)
      );

    select * from public.account;
rollback;
