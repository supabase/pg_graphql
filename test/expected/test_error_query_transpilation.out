begin;
    comment on schema public is '@graphql({"inflect_names": true})';
    create table public.account(
      id serial primary key,
      first_name varchar(255) not null
    );
    insert into public.account(first_name) values ('foo');
    -- Extend with function
    create function public._raise_err(rec public.account)
      returns text
      immutable
      strict
      language sql
    as $$
      select 1/0 -- divide by 0 error
    $$;
    select
      jsonb_pretty(
        graphql.resolve($$
          {
            accountCollection {
              edges {
                node {
                  id
                  firstName
                  raiseErr
                }
              }
            }
          }
        $$)
      );
               jsonb_pretty                
-------------------------------------------
 {                                        +
     "data": null,                        +
     "errors": [                          +
         {                                +
             "message": "division by zero"+
         }                                +
     ]                                    +
 }
(1 row)

    select * from public.account;
 id | first_name 
----+------------
  1 | foo
(1 row)

rollback;
